### PR TITLE
Lab 5 - Created IPAddr data type

### LINKED ISSUE #393: Lab 5 - Stringers
Implementation of `.String()` override for custom type `IPAddr` designed to print underlying byte slice `{0,0,0,0}` as `0.0.0.0`

### PR DESCRIPTION
**Pull request by joel00wood**
Overwrote `.String()` implementation to output underlying byte slice `{0,0,0,0}` as \'0.0.0.0\'

Fixes  #393

Review of colleague's PR #452

#### Changes proposed in this PR:
- IPAddr data type
- Overwrites underlying `.String()` to format output as an IP v4 Address

### CODE REVIEW

File: 05_stringer/joel00wood/main_test.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	}{
+		"Standard input": {
+			IPAddr{127, 0, 0, 1},
+			"127.0.0.1",
+		},
+		"googleDNS": {
+			IPAddr{8, 8, 8, 8},
+			"8.8.8.8",
+		},
+	}
```
Comment (n0npax) [inline 05_stringer/joel00wood/main_test.go:39]: please add test for empty and half-initialized `IPAddr` I mean `IPAddr{}` and `IPAddr{10.10}`

File: 05_stringer/joel00wood/main_test.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	"log"
+	"os"
+	"sync"
+	"testing"
+)
+
+// enter borrowed code
+// https://gist.github.com/hauxe/e935a7f9012bf2649710cf75af323dbf
+// stdOut capture helper
+func captureOutput(f func()) string {
```
Comment (n0npax) [inline 05_stringer/joel00wood/main_test.go]: same as: https://github.com/anz-bank/go-course/pull/359 - this is an overkill

**Review by n0npax — CHANGES_REQUESTED** [review]

File: 05_stringer/joel00wood/main_test.go (hunk 1 above)
Comment (joel00wood) [reply 05_stringer/joel00wood/main_test.go:39]: Implemented as described, thanks for pointing this out!

File: 05_stringer/joel00wood/main_test.go (hunk 2 above)
Comment (joel00wood) [reply 05_stringer/joel00wood/main_test.go]: I still partially agree but continue to find it fit for purpose

**Comment by joel00wood** [comment]
@n0npax Implemented changes where appropriate, please re-review when able :D

File: 05_stringer/joel00wood/main_test.go [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"testing"
+
+	output "github.com/joel00wood/test-helpers/capture"
+)
+
+func TestMain(t *testing.T) {
+	expected := "127.0.0.1\n"
+	actual := output.CaptureOutput(func() { main() })
+	if expected != actual {
```
Comment (n0npax) [inline 05_stringer/joel00wood/main_test.go:12]: please consider using testify instead of raw comparisions. It's not blocker in my mind.

**Review by n0npax — CHANGES_REQUESTED** [review]

**Review by n0npax — DISMISSED** [review]
LGTM

**Review by pentaphobe — DISMISSED** [review]
Good stuff, labelling as ready

(If you update your branch, please ensure you rebase!)

**Review by rokane — APPROVED** [review]

**Pull request merged**